backend/core/tests/test_api_views.py

```diff
@@ -124,3 +125,116 @@ def test_staff_users_cant_see_all_teams(self):
         url = self.get_list_url()
         response = self.client.get(url)
         self.assertEqual(len(response.data), 0)
+
+
+class TestBulkSetDueDatesApi(APITestCase, APITestCaseMixin):
+
+    LIST_URL_NAME = "team-list"
+    SUPPRESS_TEST_POST_TO_CREATE = True
+    SUPPRESS_TEST_GET_LIST = True
+
+    def verbose_instance_factory(self):
+
+        team = factories.TeamFactory()
+        user = factories.UserFactory()
+        team.user_set.add(user)
+        return team
+
+    def setUp(self):
+
+        self.blue_team = factories.TeamFactory(name='BLUE TEAM')
+        self.red_team = factories.TeamFactory(name='RED TEAM')
+        self.user_one_blue = factories.UserFactory(first_name='one_blue', is_superuser=False, is_staff=False)
+        self.user_two_blue = factories.UserFactory(first_name='two_blue', is_superuser=False, is_staff=False)
+        self.user_one_red = factories.UserFactory(first_name='one_red', is_superuser=False, is_staff=False)
+        self.user_two_red = factories.UserFactory(first_name='two_red', is_superuser=False, is_staff=False)
+        self.super_user = factories.UserFactory(first_name='super_user', is_superuser=True)
+
+        self.blue_team.user_set.add(self.user_one_blue)
+        self.blue_team.user_set.add(self.user_two_blue)
+        self.red_team.user_set.add(self.user_one_red)
+        self.red_team.user_set.add(self.user_two_red)
+
+    def test_team_members_cannot_bulk_set_due_dates_for_the_team_they_belong_to(self):
+
+        project = RecruitProjectFactory(due_time=None)
+        card = AgileCardFactory(recruit_project=project)
+        card.reviewers.add(self.user_one_red)
+        card.assignees.add(self.user_two_red)
+        self.login(self.user_two_red)
+        url = f'{self.get_instance_url(pk=self.red_team.id)}bulk_set_due_dates/'
+        response = self.client.post(path=url, format="json", data={
+            'due_time': '2021-12-03T14:17', 'content_item': str(card.content_item.id), 'team': str(self.red_team),
+            'flavour_names': 'JAVASCRIPT', 'status': card.status
+        })
+        self.assertEqual(response.status_code, 404)
+
+    def test_team_members_cannot_bulk_set_due_dates_for_teams_they_dont_belong_to(self):
+
+        project = RecruitProjectFactory(due_time=None)
+        card = AgileCardFactory(recruit_project=project)
+        card.reviewers.add(self.user_one_blue)
+        card.assignees.add(self.user_one_red)
+        self.login(self.user_one_blue)
+        url = f'{self.get_instance_url(pk=self.red_team.id)}bulk_set_due_dates/'
+        response = self.client.post(path=url, format="json", data={
+            'due_time': '2021-12-03T14:17', 'content_item': '2', 'team': str(self.red_team),
+            'flavour_names': 'JAVASCRIPT', 'status': card.status
+        })
+        self.assertEqual(response.status_code, 404)
+
+    def test_super_user_can_bulk_set_due_dates_for_a_team(self):
```

Comment:
If only a super user can set due dates then does it matter which team they belong to? I think it's enough to have `test_super_user_can_bulk_set_due_dates_for_a_team` and then one `test_non_super_user_cannot_set_due_dates` instead of the two different ones above.

Reply:
The permission is permissions=Team.PERMISSION_MANAGE_CARDS.  Therefore, correct me if I am wrong, I think that if a team member is granted access then that team member can do it.  So I am basically checking that if a team member has not been given permission that that team member cannot bulk set due dates.  I am also checking that with members from different teams.  Do you still think I should remove these tests?

Reply:
Okay cool didn't release that about the permissions.

Reply:
So it would be good to test that PERMISSION_MANAGE_CARDS works in this case.  Give someone that permission and make sure that it has the desired effect.